rootfs/api/models/app.py

```diff
@@ -385,6 +400,79 @@ def _default_structure(self, release):
 
         return structure
 
+    def verify_application_health(self, **kwargs):
+        """
+        Verify an application is healthy via the router.
+        This is only used in conjunction with the kubernetes health check system and should
+        only run after kubernetes has reported all pods as healthy
+        """
+        # Bail out early if the application is not routable
+        if not kwargs.get('routable', False):
+            return
+
+        app_type = kwargs.get('app_type')
+        self.log(
+            'Waiting for router to be ready to serve traffic to process type {}'.format(app_type),
+            level=logging.DEBUG
+        )
+
+        # Get the router host and append healthcheck path
+        url = 'http://{}:{}'.format(settings.ROUTER_HOST, settings.ROUTER_PORT)
```

Comment:
There is no way for us to verify the router has put the files into rotation and is able to respond on it without pinging the router like this.

We could possibly try to have the router try to self verify itself and update the SVC to say it's ready but @krancour would have to weigh in